### PR TITLE
Allow deleting a resource with a chart that uses a helm.sh/resource-policy annotation

### LINKED ISSUE #4713: Empty helm.sh/resource-policy annotation should not keep the resource
Output of `helm version`: v2.10.0

I was surprised that when using the stable/postgresql chart, the PVC was not destroyed when i delete the Helm release.

In this chart, the default value for "helm.sh/resource-policy" annotation is "" and not "keep" so the PVC should be destroyed.

After looking in pkg/tiller/resource_policy.go, seems that the "if resourcePolicyType == keepPolicy" should be complemented by an "else" that adds the manifest in the "remaining" array.

### PR DESCRIPTION
**Pull request by sebastien-prudhomme**
Fix  #4713 by allowing to put any value other than "keep" or empty string to delete a resource.

My use case is the PostgreSQL chart.

### CODE REVIEW

**Comment by technosophos** [comment]
NO NOT MERGE unless core maintainers reach agreement in #4713 that this is allowed. Superficially, it appears that this could result in data loss on existing implementations.

**Review by adamreese — CHANGES_REQUESTED** [review]
see note above

**Comment by sebastien-prudhomme** [comment]
@technosophos @adamreese I just add a new commit to not break compatibility and to improve the documentation on the current situation. I hope it will be accepted.

File: pkg/tiller/resource_policy.go [hunk 1]
```diff
@@ -34,28 +34,33 @@ const resourcePolicyAnno = "helm.sh/resource-policy"
...
 		resourcePolicyType = strings.ToLower(strings.TrimSpace(resourcePolicyType))
 		if resourcePolicyType == keepPolicy {
-			keep = append(keep, m)
+			toKeep = append(toKeep, m)
+		} else if resourcePolicyType == "" {
+			// Empty annotation means keep in Helm 2
+			toKeep = append(toKeep, m)
+		} else {
+			// Any other annotation means delete in Helm 2
+			toDelete = append(toDelete, m)
```
Comment (bacongobbler) [inline pkg/tiller/resource_policy.go]: I still don't think this should be accepted. The logic right in master is

1. Does the annotation `helm.sh/resource-policy` exist in the template? If yes, keep
2. Else, delete

Annotations can be added and removed from a template at any point AFAIK so why do we need to change the logic to accept `helm.sh/resource-policy: keep` to mean "delete me"? Is there some documentation somewhere or some example I can find to see why this is an issue?

**Review by bacongobbler — CHANGES_REQUESTED** [review]

**Comment by bacongobbler** [comment]
Given the amount of maintainers that have reviewed this PR and have rejected the change as well as how stale it's been sitting in the PR queue I'm going to close this for now. If you come up with an alternative PR (like the solution discussed in #4713), we'd be more than happy to take a look. Thanks!

**Pull request closed without merging**